src/base/preferences.cpp

```diff
@@ -1349,6 +1349,16 @@ QString Preferences::getNetworkInterfaceName() const
     return value("Preferences/Connection/InterfaceName").toString();
 }
 
+void Preferences::setNetworkAddress(const QString& iface)
+{
+    setValue("Preferences/Connection/InterfaceAddress", iface);
+}
+
+QString Preferences::getNetworkAddress() const
+{
+    return value("Preferences/Connection/InterfaceAddress").toString();
```

Comment:
Users might have edited the conf file and entered an invalid ip. We must validate it before using it. Try checking if a QHostAddress constructed from it is valid.